src/librustc_codegen_ironox/type_.rs

```diff
@@ -200,7 +236,13 @@ impl BaseTypeMethods<'tcx> for CodegenCx<'ll, 'tcx> {
     fn val_ty(&self, v: Value) -> Type {
         // FIXME: return the real type
         let mut borrowed_types = self.types.borrow_mut();
-        borrowed_types.push(LLType::None);
+        match v {
+            Value::ConstStruct(_) => {
+                // FIXME: implement
```

Comment:
Can this be unimplemented!? To be consistent?

Reply:
Done.

Reply:
Thanks